src/main/java/core/basesyntax/service/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.exceptions.ProductQuantityException;
+
+public class PurchaseHandler implements ActivityHandler {
+    private final Storage storage;
+
+    public PurchaseHandler(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void operate(String fruit, int quantity) {
+        int currentProductQuantity = storage.getData().get(fruit);
+
+        try {
+            int newQuantity = currentProductQuantity - quantity;
+            if (currentProductQuantity <= 0) {
```

Comment:
we should throw it before we even attempt to set it lower than 0, not after it already is, and I do not see a problem with having 0 products, would be kinda sad if you would have a single fruit at the end of each day because system does not allow to sell all of  them 😄 
```suggestion
            if (newQuantity < 0) {
```